crates/fiber-lib/src/fiber/channel.rs

```diff
@@ -5020,13 +5015,27 @@ impl ChannelActorState {
                 return Err(ProcessingChannelError::FinalIncorrectPreimage);
             }
 
-            // update balance according to the tlc
+            // update balance according to the tlc,
+            // we already checked the amount is valid in handle_add_tlc_command and handle_add_tlc_peer_message
+            // here we double confirm everything is correct.
             let (mut to_local_amount, mut to_remote_amount) =
                 (self.to_local_amount, self.to_remote_amount);
             if current.is_offered() {
+                if to_local_amount < current.amount {
+                    return Err(ProcessingChannelError::InternalError(format!(
+                        "Cannot remove tlc {:?} with amount {} from local balance {}",
+                        tlc_id, current.amount, to_local_amount
+                    )));
+                }
                 to_local_amount -= current.amount;
                 to_remote_amount += current.amount;
```

Comment:
we can use checked version ops here, its better than check amount before each ops.